openwisp_firmware_upgrader/tests/test_models.py

```diff
@@ -483,3 +484,38 @@ def test_delete_firmware_image_file(self):
             file_storage_backend.delete(file_name)
             self.assertNotEqual(image.file, None)
             image.delete()
+
+    def test_filename_within_limit(self):
+        build_object = self._create_build()
+        # Created a filename exactly at the limit of 255 characters
+        long_filename = "a" * 255
+
+        dummy_file = SimpleUploadedFile(long_filename, b"file_content")
+        firmware_image = FirmwareImage(
+            build=build_object,
+            file=dummy_file,
+            type='factory',
+        )
+        self.assertIsNotNone(firmware_image.pk)
+        try:
+            firmware_image.full_clean()
+        except ValidationError as e:
+            if 'file' in e.error_dict:
+                self.fail(f"Validation failed for filename within the limit: {str(e)}")
+        else:
+            self.assertEqual(firmware_image.file.name, long_filename)
+
+    def test_filename_exceeds_limit(self):
+        build_object = self._create_build()
+
+        # Created a filename longer than 255 characters
+        long_filename = "a" * 256
+
+        dummy_file = SimpleUploadedFile(long_filename, b"file_content")
+        firmware_image = FirmwareImage(
+            build=build_object,
+            file=dummy_file,
+            type='factory',
+        )
+        with self.assertRaises(ValidationError):
+            firmware_image.full_clean()
```

Comment:
We should ensure that the ValidationError is raised because of the filename and not any other field.

Reply:
@praptisharma28, look at other test for references, eg:
https://github.com/openwisp/openwisp-firmware-upgrader/blob/master/openwisp_firmware_upgrader/tests/test_models.py#L55-L61